### ISSUE TITLE
useQuery sometimes doesn't start polling when skip is set to true

### ISSUE THREAD

**Issue by filipcro** [apollographql/apollo-client#7667]
**Intended outcome:**

In code like this:
```
 useQuery(QUERY_FOR_POLL, {
    variables,
    pollInterval: 10000, // 10s
    skip,
  });
```

Polling should start when `skip` is set to false.

**Actual outcome:**

Sometimes polling won't start. It usually happens after starting and then stopping query.

**How to reproduce the issue:**

[Codesandbox](https://codesandbox.io/s/exciting-surf-cnffq?fontsize=14&hidenavigation=1&theme=dark) I added `console.log` to the link, so you can check if polling is working. If you click multiple times on the button, it will stop working.

**Possible workaround:**

I added `key` to the outer component which will make all hooks lose saved state, and it can work for my use case, but I think that I maybe added one edge case bug to my app.

**Versions**

```
  System:
    OS: macOS 11.1
  Binaries:
    Node: 14.13.0 - /usr/local/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
  Browsers:
    Edge: 88.0.705.63
    Safari: 14.0.2
  npmPackages:
    @apollo/client: ^3.3.0 => 3.3.1 
    apollo-upload-client: ^14.1.3 => 14.1.3 
```

**Comment by iamdiogo** [comment]
Can confirm this happens when trying to change back `skip` to `false`.

Assuming the query has some `pollInterval` value defined, toggling the value of `skip` ends up being discarded.

| number of "skip toggles"                  | 0    | 1     | 2    | 3     |
|--------------------|------|-------|------|-------|
| skip value         | true | false | true | false |
| was indeed skipped | true | false | true | true (should be false)  |

**Comment by iamdiogo** [comment]
On a first (quick) look, it seems to be related to [this line](https://github.com/apollographql/apollo-client/blob/29d41eb590157777f8a65554698fcef4d757a691/src/react/data/QueryData.ts#L64):
```javascript
    if (skip || query !== this.previous.query) {  // <-- this one
      this.removeQuerySubscription();
      this.removeObservable(!skip);
      this.previous.query = query;
    }
```

As `skip` is turned back to `false`, the query subscription and observable aren't removed. Given that `startQuerySubscription` [first checks if a subscription already exists](https://github.com/apollographql/apollo-client/blob/3a4b0d628a43cf677a24f5b18793df0ab112824d/src/react/data/QueryData.ts#L284), and returns directly if there is one, this could be the reason, as it wasn't removed by not entering the above condition.

**Comment by iamdiogo** [comment]
@jcreighton When would it be possible to have updates on this?

Having `skip` end up stop working is a bit of a deal-breaker

**Comment by FreTimmerman** [comment]
this issue still exists in v3.3.16 and v3.4.0-beta.24

**Comment by FreTimmerman** [comment]
curiously for my project, it works with this workaround:
```
const { stopPolling, startPolling } = useQuery(MY_QUERY, {
  ...
  skip: !shouldPoll,
  pollInterval: 1000,
  onError: () => {
    ...
    stopPolling(); 
    startPolling(1000);
  },
  onCompleted: () => {
    ...
    stopPolling(); 
    startPolling(1000);
  }
}
```
but this does not seem to work for the provided codesandbox example from filipcro

**Comment by glasser** [comment]
FWIW I think #6982 is the same issue and it contains a link to a test.

**Comment by glasser** [comment]
https://github.com/apollographql/apollo-client/issues/7552 might be related though it sounds like the opposite symptom.

**Comment by brainkim** [comment]
Should be fixed in 3.4. Ping me if it isn’t!
